sfge/src/main/java/com/salesforce/graph/symbols/apex/schema/DescribeSObjectResultFactory.java

```diff
@@ -3,136 +3,170 @@
-                    if (chainedNames
-                            .get(1)
-                            .equalsIgnoreCase(
-                                    ApexStandardLibraryUtil.VariableNames.S_OBJECT_TYPE)) {
-                        SObjectType sObjectType =
-                                ApexValueBuilder.getWithoutSymbolProvider()
-                                        .valueVertex(vertex)
-                                        .buildSObjectType(chainedNames.get(0));
-                        return Optional.of(
-                                ApexValueBuilder.getWithoutSymbolProvider()
-                                        .valueVertex(vertex)
-                                        .returnedFrom(sObjectType, vertex)
-                                        .buildDescribeSObjectResult(sObjectType));
-                    }
-                } else if (SystemNames.DML_OBJECT_ACCESS_METHODS.contains(methodName)
-                        && chainedNames.size() == 3) {
-                    // Schema.SObjectType.Account.isDeletable()
-                    if (chainedNames
-                                    .get(0)
-                                    .equalsIgnoreCase(ApexStandardLibraryUtil.VariableNames.SCHEMA)
-                            && chainedNames
-                                    .get(1)
-                                    .equalsIgnoreCase(
-                                            ApexStandardLibraryUtil.VariableNames.S_OBJECT_TYPE)) {
-                        SObjectType sObjectType =
-                                ApexValueBuilder.getWithoutSymbolProvider()
-                                        .valueVertex(vertex)
-                                        .buildSObjectType(chainedNames.get(2));
-                        final DescribeSObjectResult describeSObjectResult =
-                                ApexValueBuilder.getWithoutSymbolProvider()
-                                        .valueVertex(vertex)
-                                        .returnedFrom(sObjectType, vertex)
-                                        .buildDescribeSObjectResult(sObjectType);
-                        return Optional.of(
-                                ApexValueBuilder.getWithoutSymbolProvider()
-                                        .returnedFrom(describeSObjectResult, vertex)
-                                        .buildBoolean());
+                    if (methodName.equalsIgnoreCase(SObjectType.METHOD_GET_DESCRIBE)) {
+                        return getDescribeSObjectResult(vertex, sObjectName.get());
+                    } else if (SystemNames.DML_OBJECT_ACCESS_METHODS.contains(methodName)) {
+                        // Schema.SObjectType.Account.isDeletable()
+                        // SObjectType.Account.isDeletable()
+                        return getDmlAccessValue(vertex, sObjectName.get());
                     }
                 }
+
                 return Optional.empty();
             };
 
+    /**
```

Comment:
All of this logic already existed. I've only extracted them into individual methods for reusability.